PYIC-6486: Add reverification journey

MikeCollingwood opened the pull request:
## Proposed changes

### What changed

- Added mfa reset journey

### Why did it change

- To allow people to make mfa reset journeys

### Issue tracking

- [PYIC-6486](https://govukverify.atlassian.net/browse/PYIC-6486)

### Notes

- Does the entry point work with the dcmaw check?

### Verbose journey

1. Can only start if DCMAW and TICF enabled
2. page-ipv-identity-document-start (no issues here because triages if DCMAW possible)
3. Must do DCMAW (one day, STRATEGIC_APP_TRIAGE - dead end atm) otherwise INELIGIBLE > INELIGIBLE

This gives INELIGIBLE > CRI_TICF_BEFORE_ANOTHER_WAY (calls ticf) then pyi-another-way OR pyi-no-match if failed ticf

4. POST_DCMAW_SUCCESS_PAGE (or ineligible again, or failed if ci)
5. ADDRESS_AND_FRAUD (or failed: ticf, no-match)
6. EVALUATE_GPG45_SCORES (or failed: ticf, no-match)
7. CHECK_COI (or failed: ticf, no-match)
8. CRI_TICF (or FAILED_CONFIRM_DETAILS)
9. RETURN_TO_RP (or FAILED_NO_TICF)

### Notes

- I think I may need to use context for the check coi lambda because it would typically need to know the coi subjourney type, but we're not in one and we just need it to check both names and DOB

In testing:
```
https://dev-mikec.02.dev.identity.account.gov.uk/ipv/usefeatureset?featureSet=ticfCriBeta,mfaReset
```

```
https://orch-dev-mikec.02.core.dev.stubs.account.gov.uk/
```


[PYIC-6486]: https://govukverify.atlassian.net/browse/PYIC-6486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-6486: Add mfa-reset.yaml journey

MikeCollingwood committed:
PYIC-6486: Enforce ticfCriBeta enabled

MikeCollingwood committed:
PYIC-6486: Correct TICF ordering

MikeCollingwood committed:
PYIC-6486: Remove IPV_SUCCESS_PAGE state

MikeCollingwood committed:
PYIC-6486: Add TICF call if dcmaw disabled

MikeCollingwood committed:
PYIC-6486: Add and use reverification coi subjourney

MikeCollingwood committed:
PYIC-6486: Add unit tests and fix coiSubjourneyType setting

Joe-Edwards-GDS commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/reverification.yaml:
```diff
@@ -1,23 +1,189 @@
+        targetJourney: TECHNICAL_ERROR
+        targetState: ERROR_NO_TICF
+        checkFeatureFlag:
+          ticfCriBeta:
```
That doesn't look like the right feature flag!

MikeCollingwood replied:
This was intentional as the flow requires ticf & I believe you can only get onto this journey if the mfaReset feature flag is set - in IpvSessionService

Joe-Edwards-GDS replied:
What do you mean by 'requires TICF', do you mean we shouldn't support this journey if TICF is unavailable?

MikeCollingwood replied:
It looked like it was important in the MURAL design. Since I don't know when each is getting rolled out I thought that implied it was a dependency but I don't know the ordering of roll-outs so happy to make this flow inclusive

Joe-Edwards-GDS replied:
Yeah, I would treat it like the TICF CRI in all the other places.

I think it's meant to be enabled soon(TM) which will make this easier to manage!

Joe-Edwards-GDS commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/reverification.yaml:
```diff
@@ -1,23 +1,189 @@
+        checkFeatureFlag:
+          ticfCriBeta:
+            targetState: IDENTITY_START_PAGE
+        checkIfDisabled:
```
I think this should strictly be inside the `checkFeatureFlag`, although it's very unlikely to matter!

Joe-Edwards-GDS commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/reverification.yaml:
```diff
@@ -1,23 +1,189 @@
+        targetJourney: INELIGIBLE
+        targetState: INELIGIBLE
+
+  STRATEGIC_APP_TRIAGE:
```
FYI @blakeyp - this will need keeping in line with any changes in the 'main' journey flow.

I'm wondering if we should just remove all the strategic app stuff here for now, and have a ticket in the strategic app epic for rolling out across other journey types when it's closer to completion?

MikeCollingwood replied:
Will remove and add the ticket to the epic

Joe-Edwards-GDS commented on lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/reverification.yaml:
```diff
@@ -1,23 +1,189 @@
+      lambda: check-coi
+    events:
+      coi-check-passed:
+        targetState: CRI_TICF
```
We probably still need to handle TICF being disabled, unless you're sure TICF is going live first?

Joe-Edwards-GDS reviewed (commented):
This looks ok (apart from the feature flag)

It will almost certainly need updating once the UCD and new screens are confirmed, but this will be a useful starting point.

MikeCollingwood committed:
PYIC-6486:
- Nest initial dcmaw disabled check
- Remove strategic app routing
- Enable no ticf routes

Joe-Edwards-GDS reviewed (approved):
This is confirmed changing now (as mentioned on the ticket), but this will be a useful starting point and allow us to run through the journey now.
